# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1249895686 -7200
#      Mon Aug 10 11:14:46 2009 +0200
# Branch release-3-2-x
# Node ID 6ab2598a4ac23f168932d780f468606051800d74
# Parent  0cb90d7fe46ae7415cc47cb92635b2fc607a5815
parse.y (Fevalin): also return output from CATCH expression

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-08-09  John W. Eaton  <jwe@octave.org>
+
+	* parse.y (Fevalin): Also return output from CATCH expression.
+
 2009-08-06  Jaroslav Hajek  <highegg@gmail.com>
 
 	* OPERATORS/op-dm-template.cc, OPERATORS/op-pm-template.cc:
 	Also define conversions for null string and null sq_string.
 
 2009-08-06  John W. Eaton  <jwe@octave.org>
 
 	* OPERATORS/op-cdm-m.cc, OPERATORS/op-dm-m.cc, OPERATORS/op-pm-m.cc:
diff --git a/src/parse.y b/src/parse.y
--- a/src/parse.y
+++ b/src/parse.y
@@ -4275,19 +4275,20 @@ context @var{context}, which may be eith
 		{
 		  error_state = 0;
 
 		  // Set up for letting the user print any messages from
 		  // errors that occurred in the first part of this eval().
 
 		  buffer_error_messages--;
 
-		  eval_string (args(2), 0, parse_status, nargout);
-
-		  retval = octave_value_list ();
+		  tmp = eval_string (args(2), nargout > 0,
+				     parse_status, nargout);
+
+		  retval = (nargout > 0) ? tmp : octave_value_list ();
 		}
 	    }
 
 	  unwind_protect::run_frame ("Fevalin");
 	}
       else
         error ("evalin: expecting string as first argument");
     }
